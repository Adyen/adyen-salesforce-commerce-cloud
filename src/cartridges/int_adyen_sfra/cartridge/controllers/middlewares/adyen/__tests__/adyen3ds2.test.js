/* eslint-disable global-require */
const {
  adyen: { adyen3ds2 },
} = require('../../index');

let req;
let res;
let next;

beforeEach(() => {
  next = jest.fn();
  req = {
    https: true,
    host: 'mocked_host',
    querystring: {
      resultCode: 'Authorised',
      action: 'mocked_action',
      orderNo: 'mocked_orderNo',
    },
  };

  res = { redirect: jest.fn(), render: jest.fn() };
});

describe('Adyen 3DS2', () => {
  it('should render', () => {
    adyen3ds2(req, res, next);
    expect(res.render.mock.calls).toMatchSnapshot();
  });
});
