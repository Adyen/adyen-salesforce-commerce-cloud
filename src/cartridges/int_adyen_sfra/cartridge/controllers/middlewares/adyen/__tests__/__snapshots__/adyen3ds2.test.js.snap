// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adyen 3DS2 should render 1`] = `
Array [
  Array [
    "/threeds2/adyen3ds2",
    Object {
      "action": "mocked_adyen_action",
      "clientKey": "mocked_client_key",
      "environment": "test",
      "locale": "nl_NL",
      "merchantReference": "mocked_orderNo",
      "resultCode": "Authorised",
    },
  ],
]
`;
